apps/manager_app/auth_card.c

```diff
@@ -496,10 +500,16 @@ static manager_error_code_t handle_auth_card_result_query(
         return MANAGER_TASK_SUCCESS;
       } else {
         if (true == auth_card_data->ctx.pair_card_required) {
-          /* TODO: Handle this card pairing task*/
-
-          /* TODO: Check if card operation failed*/
-          // return MANAGER_TASK_FAILED;
+          uint8_t card_number =
+              decode_card_number(auth_card_data->ctx.acceptable_cards);
+          bool result = pair_card_operation(card_number,
+                                            auth_card_data->ctx.heading,
+                                            auth_card_data->ctx.message);
+
+          if (true != result) {
+            delay_scr_init("Pairing operation failed", DELAY_TIME);
```

Comment:
Can you check if having delay screen before sending response is a good idea? I think it can hamper the synchronization between cysync and device

Reply:
Agreed, working on that change separately.

Reply:
Ok, resolving for now